src/rules/no-extraneous-dependencies.js

```diff
@@ -17,6 +17,7 @@ function extractDepFields(pkg) {
     devDependencies: pkg.devDependencies || {},
     optionalDependencies: pkg.optionalDependencies || {},
     peerDependencies: pkg.peerDependencies || {},
+    bundledDependencies: pkg.bundledDependencies ||pkg.bundleDependencies || [],
```

Comment:
```suggestion
    bundledDependencies: pkg.bundledDependencies || pkg.bundleDependencies || [],
```
it's super crazy to me that both of these keys [are supported by npm](http://npm.github.io/using-pkgs-docs/package-json/types/bundleddependencies.html). Can we confirm which one takes precedence from npm itself, when both are present?

Reply:
also, if this is mistakenly defined as an object, then the code will crash later when it assumes it's an array. I think we need an `Array.isArray` check here.

Reply:
> it's super crazy to me that both of these keys [are supported by npm](http://npm.github.io/using-pkgs-docs/package-json/types/bundleddependencies.html). Can we confirm which one takes precedence from npm itself, when both are present?

Yes, I was also a bit surprised. [Looks like bundleDependencies takes precedence](https://github.com/npm/cli/blob/656bce7dd0f9753a273912e803261ed246593924/lib/install/save.js#L74) I have to change my code then.

- [ ] `bundleDependencies` takes precedence over `bundledDependencies`

> also, if this is mistakenly defined as an object, then the code will crash later when it assumes it's an array. I think we need an `Array.isArray` check here.

Good catch. On top of that, [it seems like it is also possible to define the bundleDependencies as an object](https://github.com/npm/cli/blob/c1522be2406a0ea4f14c85753edd42ddd8d7e180/test/tap/bundled-dependencies-nonarray.js#L37). Although, my VS code shows a type error (see screenshot below). Anyways, I think it would be safe to handle both cases

- [ ] Normalize array & object notation

<img width="388" alt="Screenshot 2019-08-05 at 12 38 10" src="https://user-images.githubusercontent.com/103280/62458767-0b9b4c00-b77e-11e9-962e-393ffc900ffc.png">

I'll add those fixes later today or tomorrow.